src/callbacks/get_callbacks.jl

```diff
@@ -169,9 +189,28 @@ function get_diagnostics(parsed_args, atmos_model, Y, p, dt, t_start)
         @info "$writer_str: $diags_outputs"
     end
 
-    return diagnostics, writers
+    return diagnostics, writers, periods_reductions
 end
 
+function checkpoint_frequency_from_parsed_args(dt_save_state_to_disk)
```

Comment:
Can we add a type annotation for `dt_save_state_to_disk`? This is a string, right?